kcidb/oo/__init__.py

```diff
@@ -233,6 +233,14 @@ def tests_root(self):
         """The root test node"""
         return Node(self, "")
 
+    def get_boot_tests(self, tests):
+        """Get boot tests by matching "boot" prefix in test path"""
+        boot_tests = []
+        for test in tests:
+            if test.path.split(".")[0].startswith("boot"):
+                boot_tests.append(test)
+        return boot_tests
+
```

Comment:
I don't get this.
I tried `container.tests_root["boot"].waived_status_nodes` (I assume you meant `container` as `revision` is undefined in the template) and received some kind of dictionary:
```
{False: {&#x27;FAIL&#x27;: [], &#x27;ERROR&#x27;: [], &#x27;M=
ISS&#x27;: [], &#x27;PASS&#x27;: [], &#x27;DONE&#x27;: [], &#x27;SKIP&#x27;: =
[], None: []}, True: {&#x27;FAIL&#x27;: [], &#x27;ERROR&#x27;: [], &#x27;MISS=
&#x27;: [], &#x27;PASS&#x27;: [], &#x27;DONE&#x27;: [], &#x27;SKIP&#x27;: [],=
 None: []}, None: {&#x27;FAIL&#x27;: [], &#x27;ERROR&#x27;: [], &#x27;MISS&#x=
27;: [], &#x27;PASS&#x27;: [], &#x27;DONE&#x27;: [], &#x27;SKIP&#x27;: [], No=
ne: []}}
```

Reply:
That's the same `waived_status_nodes` dictionary you were using before, in your templates, only restricted to `boot` tests and its subtests.

BTW, could you perhaps share the data you were using to test this?

Reply:
Okay, I'll check again.
I am using the same test data I have added to `test_monitor.py`.

Reply:
OK, I tried this:
```
{% set boot_tests = container.tests_root["boot"].waived_status_nodes %}
{% set failed_boot_tests = boot_tests[false]["FAIL"] %}
{{- failed_boot_tests }}
```
And that prints an empty list.

Reply:
OK, the `waived_status_nodes` would be a dictionary of waived->status->*nodes*. If you want a waived->status->*tests* for a particular node, you can apply this:
```diff
diff --git a/kcidb/oo/__init__.py b/kcidb/oo/__init__.py
index de54af1..73676ab 100644
--- a/kcidb/oo/__init__.py
+++ b/kcidb/oo/__init__.py
@@ -331,6 +331,22 @@ class Node(IncidentIssueBugContainer):
             )
         ]
 
+    @cached_property
+    def waived_status_tests(self):
+        """
+        A dictionary of all waived values and dictionaries of all status
+        values and lists of tests (test runs) with corresponding waived and
+        status values, for this and all child nodes, all in order of
+        decreasing priority.
+        """
+        waived_status_tests = {
+            waived: {status: [] for status in TEST_STATUS_PRIORITY}
+            for waived in TEST_WAIVED_PRIORITY
+        }
+        for test in self.tests:
+            waived_status_tests[test.waived][test.status].append(test)
+        return waived_status_tests
+
     @cached_property
     def incidents(self):
         """A list of incidents for this and all child nodes"""
```

And then refer to `waived_status_tests` instead of `waived_status_nodes`. E.g. this: `container.tests_root["boot"].waived_status_tests[false]["FAIL"]` will give you all non-waived failed `boot` or `boot.*` test *runs* (actual rows in the `tests` table).

If you go this route, please apply the above patch in a separate commit. Thank you!

Reply:
Thanks for the above patch.
I've added it and updated related template as well.
Should I set you as the author for the commit?

Reply:
Awesome!

> Should I set you as the author for the commit?

Don't bother :grin: